### PR TITLE
Crocodile bringup

### PR DESCRIPTION
**Pull request by htho-lgs**
Add support for crocodile

### CODE REVIEW

File: arch/arm/dts/crocodile-u-boot.dtsi [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+};
+
+&gpio5 {
+	u-boot,dm-spl;
+};
+
+&uart3 {
+	u-boot,dm-spl;
+	/delete-property/ dmas;
+	/delete-property/ dma-names;
```
Comment (zandrey) [inline arch/arm/dts/crocodile-u-boot.dtsi:70]: Please add a comment here that we do not need DMA for this UART (see example in _arch/arm/dts/crocodile.dts_ for `fec1` node).

**Review by zandrey — COMMENTED** [review]
It looks good as a base, there is just one small nitpick about the `-u-boot.dtsi` from my side.

We can use this base for further customization steps.

Nice work, @htho-lgs

**Comment by htho-lgs** [comment]
V2:
- sync crocodile.dtsi with latest https://github.com/hexagon-geo-surv/linux-leica/tree/leica/linux-5.15.y%2Fcrocodile

**Review by zandrey — APPROVED** [review]

**Comment by zandrey** [comment]
Thanks a lot, @htho-lgs!

@grembeter Please have a look and give your ACK here.

**Review by grembeter — APPROVED** [review]

**Pull request merged**